CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+#   dependencies too (e.g.
+#   https://github.com/boostorg/boost_install/issues/5). See also OTIO.
+option(BUILD_SHARED_LIBS "Set to OFF to build static libraries" ON)
+
+
+#-----------------------------------------------------------------------
+# Enable ctest testing
+
+# To correctly generate the DartConfiguration.tcl, include the CTest
+# CMake file. By doing so 'ctest -F memcheck' works.
+# TODO(DF): ...but also creates many many pointless targets.
+#include(CTest)  # Automatically calls enable_testing()
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Provides install directory variables as defined by the GNU Coding
+# Standards.
+
+include(GNUInstallDirs)
+
+#-----------------------------------------------------------------------
+# Generate OpenAssetIOConfig.cmake et al. for bundling as a CMake lib.
+
+# Targets exported to ${PROJECT_NAME}_EXPORTED_TARGETS will be available
+# for import using find_package(OpenAssetIO).
+
+include(CMakePackageConfigHelpers)
+
+set(version_config_file "${CMAKE_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config_file "${CMAKE_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+# Version fetched from the top level project()
+write_basic_package_version_file(
+    ${version_config_file}
+    # Version range supported only if generated by CMake >= 3.19.2.
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    cmake/Config.cmake.in ${project_config_file}
+    INSTALL_DESTINATION ${config_install_dir}
+    # TODO(DF): PATH_VARS?
+)
+
+install(
+    EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS
+    DESTINATION ${config_install_dir}
+    NAMESPACE ${PROJECT_NAME}::
+    FILE ${PROJECT_NAME}Targets.cmake
+)
+
+install(
+    FILES "${project_config_file}" "${version_config_file}"
+    DESTINATION "${config_install_dir}"
+)
+
+# TODO(DF): pkg-config? See OCIO.
+
+
+#-----------------------------------------------------------------------
+# Print a debug dump
+
+message(STATUS "Configuring ${PROJECT_NAME} ${PROJECT_VERSION}")
+message(STATUS "CMake ${CMAKE_VERSION}")
+message(STATUS "CMake system           = ${CMAKE_SYSTEM}")
+message(STATUS "CMake system name      = ${CMAKE_SYSTEM_NAME}")
+message(STATUS "Project source dir     = ${PROJECT_SOURCE_DIR}")
+message(STATUS "Project build dir      = ${CMAKE_BINARY_DIR}")
+message(STATUS "Project install prefix = ${CMAKE_INSTALL_PREFIX}")
+message(STATUS "Configuration types    = ${CMAKE_CONFIGURATION_TYPES}")
+message(STATUS "Build type             = ${CMAKE_BUILD_TYPE}")
+message(STATUS "Build shared libs      = ${BUILD_SHARED_LIBS}")
+
+
+#-----------------------------------------------------------------------
+# Public headers
+
+add_subdirectory(include)
+
+
+#-----------------------------------------------------------------------
+# Core C++ library
+
+add_subdirectory(src)
```

Comment:
No blank line at EOF?

Reply:
Grr, I wonder if clang-format can be configured to enforce that...

Reply:
> Grr, I wonder if clang-format can be configured to enforce that...
Duh, but this isn't C++.  Perhaps .editorconfig can help.

Reply:
It should do!

Reply:
Added blank lines and updated `.editorconfig` (though I struggle to get it to work in CLion) in f81c11d0